webapp/app/js/directives/clonejobs.js

```diff
@@ -93,6 +93,112 @@ treeherder.directive('thCloneJobs', [
     });
 
     $rootScope.$on(
+        thEvents.selectNextJob, function(ev){
+
+        var jobMap = ThResultSetModel.getJobMap($rootScope.repoName);
+        var lastJobSelected = ThResultSetModel.getSelectedJob($rootScope.repoName);
+        var el, key, job, step;
+
+        step = 'nextElementSibling';
+
+        // No job is selected currently , no job is selected till now ( 2 different cases )
+        if (lastJobSelected.el == undefined || lastJobSelected.el[0] == undefined ) {
+
+            // select first Job
+            var currentElement = document.getElementsByClassName('result-set')[0];
+            el = traceJob($(currentElement), step, 'job-btn');
+
+        } else {
+
+            var currentElement = lastJobSelected.el[0];
+
+            // Parsing for parent which satisy parameter mentioned in variable step
+            while (!currentElement[step] || $(currentElement[step]).css('display') == 'none') {
+
+                if (!currentElement[step]) {
+                    currentElement = traceParent($(currentElement), 1)[0];
+                } else {
+                    currentElement = currentElement[step];
+                }
+
+            }
+
+            // look through parent's sibling for element with class job-row
+            el = traceJob($(currentElement[step]), step, 'job-btn');
+
+        }
+
+        // reached end of elements, so element is null
+        if (el == null) {
+
+            // select first Job as a wrap up
+            var currentElement = document.getElementsByClassName('result-set')[0];
+            el = traceJob($(currentElement), step, 'job-btn');
+
+        }
+
+        key = el.attr(jobKeyAttr);
+        job = jobMap[key].job_obj;
+        selectJob(job);
+
+    });
+
+    $rootScope.$on(
+        thEvents.selectPreviousJob, function(ev){
+
```

Comment:
Nit: per my previous style guide comments in the bug - a 1 char space between the last paren and the last curly brace seem to be the general convention with a function definition. You can do the same throughout any function blocks you touch in this change.

I know there is a mix throughout the code base but we might as well aspire to the Mozilla style guide on new changes. :)